### PR TITLE
feat: add a new provider openrouter

### PR DESCRIPTION
**Pull request by LinkW77**
# Pull Request

## PR Description

Add a new provider OpenRouter

## Linked issues
#34 

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Performance enhancement
- [ ] Code refactor
- [ ] Documentation update
- [ ] Other, please describe:

## Checklist

- [x] I have read the [CONTRIBUTING.md](../CONTRIBUTING.md) guidelines.
- [x] I have tested my changes locally to ensure they are effective.
- [x] I have updated the necessary documentation (if applicable).

### CODE REVIEW

File: inference/test/utils/wildcard_test_cases.yml [hunk 1]
```diff
@@ -43,3 +51,14 @@ wildcard_test_cases:
         model_type: "text_embedding"
         properties:
           embedding_size: 999
+<<<<<<< Updated upstream
```
Comment (jameszyao) [inline inference/test/utils/wildcard_test_cases.yml:54]: @LinkW77 solve conflicts first

File: inference/test/test_chat_completion.py [hunk 2]
```diff
@@ -233,7 +233,11 @@ async def test_chat_completion_by_stream_and_function_call(self, test_data):
         message = test_data["message"]
         function_call = test_data["function_call"]
         stream = test_data["stream"]
+<<<<<<< Updated upstream
         if not function_call or not stream or "azure" in model_schema_id:
```
Comment (jameszyao) [inline inference/test/test_chat_completion.py:237]: @LinkW77 resolve conflicts first

**Review by jameszyao — DISMISSED** [review]
LGTM

**Review by SimsonW — DISMISSED** [review]
LGTM

**Review by SimsonW — APPROVED** [review]
LGTM

**Review by jameszyao — APPROVED** [review]
LGTM

**Pull request merged**